### DISCUSSION TITLE
Possibility to pass extra selectors to createCustomFocusIndicatorStyle

### DISCUSSION THREAD

**Discussion by flora8984461** [microsoft/fluentui#25595] [Ideas] [answered]
According to the [createCustomFocusIndicatorStyle](https://github.com/microsoft/fluentui/blob/09b24f994d3bb0aa7c072f93430a60fe23076c3d/packages/react-components/react-tabster/src/focus/createCustomFocusIndicatorStyle.ts#L27)

Can it allow adding children elements selectors and / or other elements selectors?  

For example, 
User passes params like this:
```
const children = "div" / ".classname" / "#someid" / "div .classname #someid" / etc.
createCustomFocusIndicatorStyle(style, { selector: 'focus' }, children)
```

And the implementation of the function can be like:
```
export const createCustomFocusIndicatorStyle = (
  style: GriffelStyle,
  {
    selector = defaultOptions.selector,
    enableOutline = false,
  }: CreateCustomFocusIndicatorStyleOptions = defaultOptions,
): GriffelStyle => ({
  `:focus ${children}`: {
    outlineStyle: enableOutline ? undefined : 'none',
  },
  ':focus-visible ${children}': {
    outlineStyle: enableOutline ? undefined : 'none',
  },

  ...(selector === 'focus' && {
    [`&[${FOCUS_VISIBLE_ATTR}] ${children}`]: style,
  }),
  ...(selector === 'focus-within' && {
    [`&[${FOCUS_WITHIN_ATTR}]:${selector} ${children}`]: style,
  }),
});
```

And use"+" sign so that the focus style can be applied to other elements as well.
```
const extra = "div" / ".classname" / "#someid" / "div .classname #someid" / etc.

export const createCustomFocusIndicatorStyle = (
  style: GriffelStyle,
  {
    selector = defaultOptions.selector,
    enableOutline = false,
  }: CreateCustomFocusIndicatorStyleOptions = defaultOptions,
): GriffelStyle => ({
  `:focus+${extra}`: {
    outlineStyle: enableOutline ? undefined : 'none',
  },
  ':focus-visible+${extra}': {
    outlineStyle: enableOutline ? undefined : 'none',
  },

  ...(selector === 'focus' && {
    [`&[${FOCUS_VISIBLE_ATTR}]:focus+${extra}`]: style,
  }),
  ...(selector === 'focus-within' && {
    [`&[${FOCUS_WITHIN_ATTR}]:${selector}+${extra}`]: style,
  }),
});
```

[Example sandbox](https://codesandbox.io/s/blazing-fire-2w6tvk?file=/example.tsx)

Although I can add the styles directly, and not to use `createCustomFocusIndicatorStyle` or `createFocusOutlineStyle`, I am curious about the thoughts here to add into the functions for convenience. 

Thank you in advance for sharing any insights here. 😸

**Comment by ling1726** [comment, marked as answer]
Would there be any difference compared to using an ampersand selector? @flora8984461 

https://codesandbox.io/s/eloquent-pond-6ej5nm?file=/example.tsx

**Reply by flora8984461** [reply]
Thank you for help 🙏 
Didn't realize we can use ampersand selector, this would work!
